contracts/libraries/LockDataLibrary.sol

```diff
@@ -2,49 +2,92 @@
 pragma solidity ^0.8.20;
 
 import {IPoolManager} from "../interfaces/IPoolManager.sol";
+import {LockData} from "../types/LockData.sol";
 
 /// @dev This library manages a custom storage implementation for a queue
-///      that tracks current lockers. The "sentinel" storage slot for this data structure,
-///      always passed in as IPoolManager.LockData storage self, stores not just the current
+///      that tracks current lockers. The LOCK_DATA storage slot for this data structure,
+///      passed in as a LockData custom type that is a uint256, stores not just the current
 ///      length of the queue but also the global count of non-zero deltas across all lockers.
-///      The values of the data structure start at OFFSET, and each value is a locker address.
+///      The values of the data structure start at LOCKERS, and each value is a locker address.
 library LockDataLibrary {
-    uint256 private constant OFFSET = uint256(keccak256("LockData"));
+    using LockDataLibrary for LockData;
 
-    /// @dev Pushes a locker onto the end of the queue, and updates the sentinel storage slot.
-    function push(IPoolManager.LockData storage self, address locker) internal {
-        // read current value from the sentinel storage slot
-        uint128 length = self.length;
+    uint256 private constant LOCK_DATA = uint256(keccak256("LockData"));
+    uint256 private constant LOCKERS = uint256(keccak256("Lockers"));
+
+    /// @dev Pushes a locker onto the end of the queue at the LOCKERS + length storage slot, and updates the length at the LOCK_DATA storage slot.
+    function push(LockData lockData, address locker) internal {
+        // read current length from the LOCK_DATA storage slot
+        uint128 _length = lockData.length();
         unchecked {
-            uint256 indexToWrite = OFFSET + length; // not in assembly because OFFSET is in the library scope
+            uint256 lockerSlot = LOCKERS + _length; // not in assembly because LOCKERS is in the library scope
+
             /// @solidity memory-safe-assembly
             assembly {
                 // in the next storage slot, write the locker
-                sstore(indexToWrite, locker)
+                sstore(lockerSlot, locker)
             }
-            // update the sentinel storage slot
-            self.length = length + 1;
+            lockData.update(_length + 1, lockData.nonzeroDeltaCount());
         }
     }
 
     /// @dev Pops a locker off the end of the queue. Note that no storage gets cleared.
-    function pop(IPoolManager.LockData storage self) internal {
+    function pop(LockData lockData) internal {
+        uint128 _length = lockData.length();
```

Comment:
It doesn't need to save the length to a temporary variable.
`lockData.length()` can be used directly in the below code